### ISSUE TITLE
Error adding partition as an OSD

### ISSUE THREAD

**Issue by Sensei-CHO** [canonical/microceph#68]
I'm trying to add a partition from one of my disks to microceph.

I tried with ext4 fromat, and unformated.

`microceph disk add /dev/sda3`

```
Error: Failed adding new disk: Failed to bootstrap OSD: Failed to run: ceph-osd --mkfs --no-mon-config -i 0: exit status 250 (2022-12-12T18:43:58.421+0000 7fc07269a5c0 -1 bluestore(/var/lib/ceph/osd/ceph-0/block) _read_bdev_label failed to open /var/lib/ceph/osd/ceph-0/block: (13) Permission denied
2022-12-12T18:43:58.421+0000 7fc07269a5c0 -1 bluestore(/var/lib/ceph/osd/ceph-0/block) _read_bdev_label failed to open /var/lib/ceph/osd/ceph-0/block: (13) Permission denied
2022-12-12T18:43:58.421+0000 7fc07269a5c0 -1 bluestore(/var/lib/ceph/osd/ceph-0/block) _read_bdev_label failed to open /var/lib/ceph/osd/ceph-0/block: (13) Permission denied)
```
I tried to run the command with `sudo` and as root but nothing worked and `/var/lib/ceph/osd/ceph-0/block`(neither `/var/lib/ceph/`) do not exist (shouldn't it be `/var/snap/microceph/common/data/osd/ceph-0` ?)

**Comment by sabaini** [comment]
You will need to run with sudo, like `sudo microceph disk add /dev/sda3`. Also, in case the partition has had any data recorded previously it might be prudent to add wipe: `sudo microceph disk add /dev/sda3 --wipe` 

The data dirs for Ceph are located under `/var/snap/microceph/common/data/osd/ceph-N` (for some OSD number N). The error message from ceph-osd utility is a bit misleading here as it's reporting the filesys structure as the snap sees it, which is mounted onto the host filesystem. 

If it fails with sudo please post the error message you get and possibly also recent /var/log/syslog lines

**Comment by stgraber** [comment]
The actual wiping process is done by the daemon, so `sudo` shouldn't matter so long as you're allowed to talk to the microceph socket.

**Comment by stgraber** [comment]
Which to be fair is normally restricted to the root user, so you do need sudo, just for a different reason ;)

**Comment by tonysmithio** [comment]
![microceph_init_err](https://user-images.githubusercontent.com/69742294/230912648-9a17e02f-af31-456e-9b2a-520403c8d949.png)

@stgraber

**Comment by sabaini** [comment]
@tonysmithio this looks like the issues fixed by #110 / #114  -- can you retry with the latest microceph version?

**Comment by tonysmithio** [comment]
Sorry for the late reply. I switched to the edge/latest channel and
it worked. Thanks for your help
